test/DomainRegistry.js

```diff
@@ -0,0 +1,87 @@
+const { expect } = require("chai");
+
+describe("DomainRegistry", function () {
+  let DomainRegistry, domainRegistry, owner, addr1, addr2;
+  const initialDeposit = ethers.parseEther("1");
+
+  beforeEach(async () => {
+    DomainRegistry = await ethers.getContractFactory("DomainRegistry");
+    [owner, addr1, addr2] = await ethers.getSigners();
+    domainRegistry = await DomainRegistry.deploy(initialDeposit);
+  });
+
+  it("Should deploy properly", async function () {
+    expect(await domainRegistry.getTotalReservedDomains()).to.equal(0);
```

Comment:
Why no test for getTotalReservedDomains not equal to 0? What if it is 0 all the time or is wrong in some cases.
-1 point